homeassistant/components/lutron/fan.py

```diff
@@ -0,0 +1,69 @@
+"""Support for Lutron fans."""
+
+from homeassistant.components.fan import DOMAIN, SUPPORT_SET_SPEED, FanEntity
+from . import DOMAIN as LUTRON_DOMAIN, LUTRON_CONTROLLER, LUTRON_DEVICES, LutronDevice
+
+
+DEFAULT_ON_PERCENTAGE = 50
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None) -> None:
+    """Set up the Lutron fans."""
+    devs = []
+
+    # Add Lutron Fans
+    for (area_name, device) in hass.data[LUTRON_DEVICES][DOMAIN]:
+        dev = LutronFan(area_name, device, hass.data[LUTRON_CONTROLLER])
+        devs.append(dev)
+
+    add_entities(devs, True)
+
+
+class LutronFan(LutronDevice, FanEntity):
+    """Representation of a Lutron Fan controller. Including Fan Speed."""
+
+    def __init__(self, area_name, lutron_device, controller):
+        """Initialize the fan device."""
+        self._prev_speed = None
```

Comment:
Looks like `_prev_speed` is set here but then never used. Suspect this is an artifact of copy/pasta from https://github.com/home-assistant/core/blob/dev/homeassistant/components/lutron/light.py#L36.

Rather than cleaning it up, I would actually recommend, sticking with this variable and using it below in the the `turn_on(...)` method where if a percentage is not provided, it will set the previous speed.  From an end user experience, turning off then on the fan via HA, will not cause the fan to lose its previous speed and go back to the `DEFAULT_ON_PERCENTAGE`.

For reference I had actually implemented a duplicate [PR](https://github.com/home-assistant/core/pull/56959/files#diff-80e24b982558092127a33e624eb8654a54980052f8b35a84ec5de989892a1e7fR22) using this approach as it aligned with how the dimmer/light entity has been implemented.